compiler/src/dotty/tools/dotc/util/SourcePosition.scala

```diff
@@ -70,7 +70,7 @@ extends interfaces.SourcePosition with Showable {
 }
 
 /** A sentinel for a non-existing source position */
-@sharable object NoSourcePosition extends SourcePosition(NoSource, NoSpan) {
+@sharable object NoSourcePosition extends SourcePosition(NoSource, NoSpan, null) {
```

Comment:
How come this code worked in Scala 2 ? Can we try to emulate their behavior ?

Reply:
Scala 2 behavior is incorrect, as the test case `tests/run/i4659b.scala` shows.

Reply:
How is it incorrect ? `tests/run/i4659b.scala` compiles and runs without errors on Scala 2.12.8

Reply:
In `tests/run/i4659b.scala`, `NoSourcePosition` is used as a default value before it is initialized.   It should not compile.

While rejecting `tests/run/i4659b.scala` is currently out of reach, we can do better than Scala2 by rejecting `tests/neg/i4659b.scala` .

Reply:
I see, then we should leave a comment in `tests/run/i4659b.scala` to say that we'd like this to not compile eventually.